### PR TITLE
Improve logs gathering for openstack-k8s-operators

### PR DESCRIPTION
**Pull request by viroel**

### CODE REVIEW

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-openstack-operator-master-openstack-operator-build-deploy-tempest

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-openstack-operator-master-openstack-operator-build-deploy-tempest

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-openstack-operator-master-openstack-operator-build-deploy-tempest

**Comment by rlandy** [comment]
Looks ok from checking artifacts: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/37084/rehearse-37084-pull-ci-openstack-k8s-operators-openstack-operator-master-openstack-operator-build-deploy-tempest/1633583806369763328/artifacts/openstack-operator-build-deploy-tempest/openstack-k8s-operators-gather/artifacts/

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-openstack-operator-master-openstack-operator-build-deploy-tempest

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-openstack-operator-master-openstack-operator-build-deploy-tempest

**Comment by viroel** [comment]
now with OCP must-gather:
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/37084/rehearse-37084-pull-ci-openstack-k8s-operators-openstack-operator-master-openstack-operator-build-deploy-tempest/1633923915585687552/artifacts/openstack-operator-build-deploy-tempest/openstack-k8s-operators-gather/artifacts/

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-openstack-operator-master-openstack-operator-build-deploy-tempest

tempest failed, not related with this patch

File: ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh
Comment (marios) [reply ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh:6]: oh alabama is the default ... so /alabama is meant to just fail?

Comment (marios) [reply ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh:6]: and it would be pretty bad if $HOME is not set so ... i see

File: ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh
Comment (marios) [inline ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh:26]: nice thank you is much better like this for future generations of oooci engineers ;)

**Review by marios — COMMENTED** [review]
looks good

failed job is happening during tempest so not related to this afaics 

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/37084/rehearse-37084-pull-ci-openstack-k8s-operators-openstack-operator-master-openstack-operator-build-deploy-tempest/1633970107900235776/build-log.txt

{0} setUpClass (tempest.api.identity.admin.v3.test_credentials.CredentialsTestJSON) [0.000000s] ... FAILED

File: ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh
Comment (marios) [reply ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh:6]: so apparently /alabama must exist ? it is used in the job @ 
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/37084/rehearse-37084-pull-ci-openstack-k8s-operators-openstack-operator-master-openstack-operator-build-deploy-tempest/1633970107900235776/artifacts/openstack-operator-build-deploy-tempest/openstack-k8s-operators-gather/build-log.txt


```
+ BASE_DIR=/alabama
+ MUST_GATHER_TIMEOUT=5m
+ unset NAMESPACE
+ oc project openstack
Now using project "openstack" on server "https://api.oko-02.oooci.ccitredhat.com:6443".
+ pushd /alabama
```

Comment (marios) [reply ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh:6]: @viroel do you have any more context about /alamaba please :)

File: ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh
Comment (Sandeepyadav93) [inline ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh:52]: For Openshift core operators, pods information [1] is collected in must gather itself but looks like this is not available for OLM operators reading[2]  and [3].

Looks like we need to manually collect pods information out of must-gather.

[1] https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/37084/rehearse-37084-pull-ci-openstack-k8s-operators-openstack-operator-master-openstack-operator-build-deploy-tempest/1633970107900235776/artifacts/openstack-operator-build-deploy-tempest/openstack-k8s-operators-gather/artifacts/must-gather/quay-io-openshift-release-dev-ocp-v4-0-art-dev-sha256-0291e5ed752e993177e25c1291484b63bc5ab38bd8d6f0f8020d78c8bc6cd5f6/namespaces/openshift-dns/pods/

[2] https://github.com/openshift/must-gather/issues/319
[3] https://github.com/openshift/must-gather/issues/247

File: ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh
Comment (viroel) [reply ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh:6]: Sorry, I left my last coment in Pending here. 
/alabama dir seems to be used by "ci-operator" tool as default $HOME dir when runnig multi-step jobs.
So HOME should always be "/alabama" unless we set another $HOME value (which needs to be included in step definition).
I didn't find anything in docs, only in their code [1]

[1] https://github.com/openshift/ci-tools/blob/66cded6a7970c2c05a43fb48f3e33c9b97533593/pkg/steps/multi_stage/gen.go#L168-L169

File: ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh
Comment (viroel) [reply ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh:26]: :)

File: ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh
Comment (viroel) [inline ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh:57]: So do we want to continue with must-gather? We can add a boolean to skip this, so we can have this code ready to run when dev need it.

Comment (Sandeepyadav93) [reply ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh:57]: I think it still makes sense to capture must gather to check openshift logs - its possible that openshift itself can have issues, so must gather will allow us to debug issues in openshift.

**Comment by rlandy** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-openstack-operator-master-openstack-operator-build-deploy-tempest

**Comment by Sandeepyadav93** [comment]
This looks good to me:

Operator logs:
===========

* controller manager logs:

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/37084/rehearse-37084-pull-ci-openstack-k8s-operators-openstack-operator-master-openstack-operator-build-deploy-tempest/1635965471515742208/artifacts/openstack-operator-build-deploy-tempest/openstack-k8s-operators-gather/artifacts/controller-manager/

* resources logs:

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/37084/rehearse-37084-pull-ci-openstack-k8s-operators-openstack-operator-master-openstack-operator-build-deploy-tempest/1635965471515742208/artifacts/openstack-operator-build-deploy-tempest/openstack-k8s-operators-gather/artifacts/resources/

* operator/install plan/operator group/subscription in must gather:
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/37084/rehearse-37084-pull-ci-openstack-k8s-operators-openstack-operator-master-openstack-operator-build-deploy-tempest/1635965471515742208/artifacts/openstack-operator-build-deploy-tempest/openstack-k8s-operators-gather/artifacts/must-gather/quay-io-openshift-release-dev-ocp-v4-0-art-dev-sha256-0291e5ed752e993177e25c1291484b63bc5ab38bd8d6f0f8020d78c8bc6cd5f6/namespaces/openstack/operators.coreos.com/


Openstack Services logs:
===================

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/37084/rehearse-37084-pull-ci-openstack-k8s-operators-openstack-operator-master-openstack-operator-build-deploy-tempest/1635965471515742208/artifacts/openstack-operator-build-deploy-tempest/openstack-k8s-operators-gather/artifacts/services/

Openshift logs:
============

* Openshift logs to debug issues in Openshift side captured in must gather:

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/37084/rehearse-37084-pull-ci-openstack-k8s-operators-openstack-operator-master-openstack-operator-build-deploy-tempest/1635965471515742208/artifacts/openstack-operator-build-deploy-tempest/openstack-k8s-operators-gather/artifacts/must-gather/quay-io-openshift-release-dev-ocp-v4-0-art-dev-sha256-0291e5ed752e993177e25c1291484b63bc5ab38bd8d6f0f8020d78c8bc6cd5f6/

I will give others a chance to give their input as well before we merge this.

File: ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh
Comment (stuggi) [inline ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh:24]: should this be a `.yaml` instead of `.log`

File: ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh
Comment (viroel) [reply ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh:57]: @stuggi agree with you. Since we still don't have this image, make sense we use the default one here for now, and replace with ours once is ready?

File: ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh
Comment (viroel) [reply ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh:44]: Ack, this will affect all steps in the end. But I can already parameterize both here, operators-ns and services-ns. I will provide an update for that.

File: ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh
Comment (viroel) [reply ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh:24]: make totally sense, i will update that

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-openstack-operator-master-openstack-operator-build-deploy-tempest

File: ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh
Comment (stuggi) [inline ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh:56]: - we should also collect details of  all  services, routes, deployments, daemonset, replicaset. 

could just loop over 
`oc get all -o name -n "${NS_SERVICES}" `
and then dump the yamls for each

- Also should collect all NADS of the namespace (only exist in network isolation)

`oc get network-attachment-definition -n "${NS_SERVICES}"`

- and the nncps (only exist in network isolation)
`oc get nncp`

- metallb resource (only exist in network isolation)
`oc get ipaddresspool -n metallb-system`
`oc get l2advertisement -n metallb-system`

Comment (viroel) [reply ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh:56]: Thanks for the inputs, I will work on that too.

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-openstack-operator-master-openstack-operator-build-deploy-tempest

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-openstack-operator-master-openstack-operator-build-deploy-tempest

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-openstack-operator-master-openstack-operator-build-deploy-tempest

File: ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh
Comment (viroel) [reply ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh:102]: Added, some outputs here[1]

[1] https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/37084/rehearse-37084-pull-ci-openstack-k8s-operators-openstack-operator-master-openstack-operator-build-deploy-tempest/1636161182786129920/artifacts/openstack-operator-build-deploy-tempest/openstack-k8s-operators-gather/artifacts/

File: ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh
Comment (gibizer) [inline ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh:27]: nit: Can we instruct `oc logs` to somehow separate logs from different containers in the same Pod in the output. I only found `--prefix`, that would include the name of the container into the output. For me a single empty line or any other clear separator would be enough between the logs though. If that is not possible then I would like to have --prefix added to the command.

File: ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh
Comment (gibizer) [inline ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh:94]: Do I understand correclty that we collect all the service pod logs and then we collect again the ones in error so that we can highlight the error ones? (I'm OK with this approach)

**Review by gibizer — COMMENTED** [review]
This is a really good enhancement. Kudos for implementing it. I have only nits and questions inline. None of which warrant holding this up as they can be fixed in a follow up PR.

File: ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh
Comment (viroel) [reply ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh:67]: Yeah, I'm using it here just to use its replace `-I` . I could use `sed` instead, but it was giving headaches since ${i} contains slash keys. This was simpler

File: ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh
Comment (viroel) [reply ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh]: Good idea, I vote for using `--prefix` for now. We may want to improve that in a next PR if needed.

File: ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh
Comment (viroel) [reply ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh:90]: thanks for raising that point.

File: ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh
Comment (viroel) [reply ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh:94]: Yes, I'm just doing this because that logic was already here, someone added this in a previous commit. I think that might help on finding issues too.

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-openstack-operator-master-openstack-operator-build-deploy-tempest

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-openstack-operator-master-openstack-operator-build-deploy-tempest

**Review by stuggi — APPROVED** [review]
/lgtm

**Pull request merged**
